pymeasure/instruments/hp/hp856Xx.py

```diff
@@ -2568,9 +2564,18 @@ def store_thru(self):
             When a trace is in max-hold mode, if the threshold is raised above any of the
             trace data, the data below the threshold will be permanently lost.
         """,
-        validator=joined_validators(strict_discrete_set, strict_range),
-        values=[["ON", "OFF"], arange(-200, 30)],
-        set_process=lambda v: v if isinstance(v, str) else ("%.2E {amplitude_unit}" % v)
+        validator=strict_range,
+        values=arange(-200, 30),
```

Comment:
I'm not sure, whether that's the right combination of validator and values, but I understand, that it is not the scope of your PR.
AFAIK, strict_range just needs two values (`values=[1, 999]`) and not a list or iterator as `arange` is.

Reply:
It more or less is the right way, because it is needed that you can't set 100.5

Reply:
in that case, you need `strict_discrete_set` as a validator, because the `strict_range` does allow anything between min and max of the list